extensions/nanoarrow_ipc/src/nanoarrow_ipc/nanoarrow_ipc.c

```diff
@@ -0,0 +1,352 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <errno.h>
+#include <string.h>
+
+#include "nanoarrow/nanoarrow.h"
+
+#include "File_reader.h"
+#include "Message_reader.h"
+#include "Message_verifier.h"
+#include "Schema_reader.h"
+
+#include "nanoarrow_ipc.h"
+
+#define ArrowIpcErrorSet(err, ...) ArrowErrorSet((struct ArrowError*)err, __VA_ARGS__)
```

Comment:
Hmm, is this actually useful? (Also technically, I think you're not supposed to pun between two structs just because they happen to have the same fields...)

What do you think about having nanoarrow_ipc.h just include nanoarrow.h, and just reuse existing utilities?

Reply:
Ah, I guess the question is, is this a separately library using nanoarrow internally, or is it just an optional part of nanoarrow?

I don't mind this not exposing nanoarrow. But this is technically undefined behavior here, as I understand.